tutorials/commands/flux-mini-alloc.rst

```diff
@@ -0,0 +1,251 @@
+
+Just like above, lets submit a job to this subinstance, but we'll name the job "Level2".
+
+.. code-block:: console
+
+   corona173:~$ flux mini submit --job-name=Level2 -N1 sleep inf
+
+   corona173:~$ flux jobs
+           JOBID USER     NAME       ST NTASKS NNODES     TIME INFO
+        ƒb63wFg3 achu     Level2      R      1      1   10.68s corona174
+
+   corona173:~$ flux resource list
+         STATE NNODES   NCORES    NGPUS NODELIST
+          free      1       48        8 corona173
+     allocated      1       48        8 corona174
+          down      0        0        0
+
+As you can see, another job has been submitted and has taken up a resource of one node.  You'll also notice
+that the "Level1" sleep job is not listed.  The reason is because that job is not a part of this Flux instance,
+it is in this subinstance's parent.
+
+Lets take a look at this at a global level.  Lets go back to the login node and run ``flux jobs``.
+
+.. code-block:: console
+
+    corona-login-node:~$ flux jobs
+           JOBID USER     NAME       ST NTASKS NNODES     TIME INFO
+     ƒgpD9HY9BsM achu     flux        R      4      4   3.997m corona[171,173-175]
+
+We ony see one job, which is our subinstance.  Where is our other subinstance?  Where are our two sleep jobs?
+
+.. note::
+
+   Depending on your terminal settings, the subinstance may be colored blue in ``flux jobs``, indicating it is a subinstance.
+
+We can see them by specifying the ``--recursive`` option to ``flux-jobs.
+
+.. code-block:: console
+
+    corona-login-node:~$ flux jobs --recursive
+           JOBID USER     NAME       ST NTASKS NNODES     TIME INFO
+     ƒgpD9HY9BsM achu     flux        R      4      4    4.74m corona[171,173-175]
+
+    ƒgpD9HY9BsM:
+        ƒge9VDjD achu     flux        R      2      2   3.199m corona[173-174]
+        ƒLU5z2zj achu     Level1      R      1      1   3.961m corona175
+
+    ƒgpD9HY9BsM/ƒge9VDjD:
+        ƒb63wFg3 achu     Level2      R      1      1   1.903m corona174
+
+From this output we can see that our first subinstance has two jobs, our
+"Level1" sleep job and a subinstance in it.  Our second subinstance has one job
+in it, our "Level2" sleep job.
+
+Another way to view this hierarchy of subinstances is via the ``flux-pstree`` command.
+
+.. code-block:: console
+
+    corona-login-node:~$ flux pstree
+    .
+    └── flux
+        ├── flux
+        │   └── Level2
+        └── Level1
+
+-------------------------------
+Subinstance via Flux Mini Batch
+-------------------------------
+
+``flux mini alloc`` is great for when you need to drop into a shell to interact
+with your subinstance.  This is often used by users to drop into a "main"
+subinstance and to interact with that subinstance.
+
+However, the majority of the time you wouldn't want this.  Most often, you want
+to launch a subinstance, perhaps launch a number of jobs within those
+subinstances, and just wait for them to complete.  The most common way to do
```

Comment:
Can you give a workflow / real world use case of wanting to do this (I'm trying to understand myself still).

Reply:
Dunno if you're asking specifically about `batch` vs `alloc` for this question, but basically most of the time users don't want to be dropped into a `shell`, they just want to launch their job and walk away.  I can perhaps clear that up more here.

Reply:
Why do I want to launch a subinstance? I'm dense, I don't get it. Am I running a scientific workflow that is going to launch to GPU nodes? Am I writing a python script? A bash script? I don't get the use case - the alloc sort of demonstrates the concept of a subinstance, but in practice I don't know why I'd want to then use batch. What am I doing?